master/buildbot/schedulers/basic.py

```diff
@@ -71,8 +73,9 @@ def __init__(self, name, shouldntBeSet=NotSet, treeStableTimer=None,
 
         # the IDelayedCall used to wake up when this scheduler's
         # treeStableTimer expires.
-        self._stable_timers = defaultdict(lambda: None)
-        self._stable_timers_lock = defer.DeferredLock()
+        self._stable_timers = {}
```

Comment:
There's not much need for a defaultdict here. It's better to explicitly test what's in the dict. Plus I wanted the two timers to work similarly and dict was easier for cool-down.